### ISSUE TITLE
Use an iso to mount the repos to the hosts instead of http

### ISSUE THREAD

**Issue by david-caro** [lago-project/lago#55]
Moved from https://bugzilla.redhat.com/show_bug.cgi?id=1278552

Right now the locally generated repo is passed to the hosts through http, that requires lago to start a process serving it on http, it would be easier and avoid port conflicts and allow using it offline to mount it on the hosts through an iso (or whatever)

**Comment by mykaul** [comment]
I'm not sure I see the benefit - it requires creating the ISO in the first place, no? This takes time and disk space.
Moreover, I'd be happy if we could use shared (NFS) repos to serve from. Currently we are not able to do so, as the merge actually copies the RPMs if it cannot hard-link them. Not sure why.

**Comment by david-caro** [comment]
Using shared nfs for the repos does defeat the whole point of pulling them locally no? We can just use the proxy to cache them all instead.

Using an iso instead of using http+port on the local machine will remove all and any network issues and similar, and allow faster installations (pulling from local iso is way faster than using an intermediate dummy http server).

But yes, you will have to create the iso and that's a big drawback.

About the rpm copies, can you elaborate? I don't know what are you referring to

**Comment by mykaul** [comment]
Network is needed anyway for reposync, no? Unless we can have an option (which I think is very valid) to skip reposync.

Using shared NFS is great if our RPMs can be shared and served from a single location (which is probably faster than your local disk, unless you have SSD). You are then not copying it from it, only symlinking... 

The code doing the copy is in contrib/ovirt/lib/ovirtlago/merge_repos.py :
37 def _fastcopy(source, dest):
 38     try:
 39         os.link(source, dest)
 40     except OSError:
 41         shutil.copy(source, dest)


It should do os.symlink() to support this over NFS.

**Comment by david-caro** [comment]
Yes, that is done to avoid copying over the rpms on the same hard drive if it can be avoided, when creating the final internal_repo dir inside the prefix, and avoid leaving the prefix unusable if you update the caches and changes are pulled in.

I don't think it's a good idea to use nfs, it complicates things too much the setup and does not add any value to the developer (that will run this on her laptop, and expects not to need anything external once the prefix is initialized).

If the goal is to speed up the runs on oVirt CI, I vote for the network proxy, maybe add to lago the option to use a proxy during reposync so you don't have to hard-code the proxy url in the config. It should be quite similar to nfs performance, does not have any locking/coordination issues, does not require any new logic to work, and it's already being used for ci jobs (the proxy I mean).

**Comment by mykaul** [comment]
To be honest, I did not need it for NFS. I needed it for zram based mounts. Later on I've just used /dev/shm ....

The goal is indeed clearly to do whatever we can to accelerate CI. I've managed to cut it down to 18 minutes (from initial 27) on my desktop, but still working on some more ideas.

**Comment by david-caro** [comment]
Nice! Can you elaborate a bit on how you are doing that? Maybe we can help :smile:

**Comment by mykaul** [comment]
Fix ./run_suite.sh as I've proposed before:
@@ -174,7 +174,9 @@ if [[ -z "$1" ]]; then
 fi

 export SUITE="$(realpath "$1")"
-export PREFIX="$PWD/deployment-${SUITE##*/}"
+if [ -z "$PREFIX" ]; then
+    export PREFIX="$PWD/deployment-${SUITE##*/}"
+fi



And then run:
mkdir /dev/shm/3.6
/run_suite.sh  -o /dev/shm/3.6 basic_suite_3.6


Pray you have enough RAM!

**Comment by david-caro** [comment]
definitely that will help on the ci env
